library/CHART-TEMPLATE/templates/deployment.yaml

```diff
@@ -49,6 +49,11 @@ spec:
               port: http
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
+          env:
+            - name: MY_APP_NON_OPTIONAL_ENV_VAR
```

Comment:
why no value for the env? 


in nri-prometheus for example the approach is different:

https://github.com/newrelic/helm-charts/blob/afd566c86e2d336c08d52edc7ab5a60fae6859b7/charts/nri-prometheus/values.yaml#L73



The user is required to add the url, it is not enough just to enable a flag

Reply:
but in some other apps there are more than 1 endpoint. for instance the agent has 3 endpoints. And even if they would have just one endpoint to configure they are usually different (meaning DM endpoint, Otel endpoint , ... )
This is why i followed the boolean approach

Reply:
I also though about something like this 
```yaml
fedramp:
  enabled: true
  metrics_api_url:
  events_api_url:
  ...
```
but it would required some internals knowledge of each component that we could save from users i think.

Reply:
So the idea would be to have in global just the enable and in each chart the url value?

Reply:
yes so instead of configuring each endpoint manually in each subchart , just enabled this flag and all subchart could set this for you with the default values or [flags](https://docs.newrelic.com/docs/infrastructure/install-infrastructure-agent/configuration/infrastructure-agent-configuration-settings#fedramp) depending on the sub chart .

Reply:
ok, makes sense!